thrust/thrust/system/cuda/detail/sort.h

```diff
@@ -299,7 +299,14 @@ namespace __smart_sort
 template <class Key, class CompareOp>
 using can_use_primitive_sort = ::cuda::std::integral_constant<
   bool,
-  ::cuda::std::is_arithmetic<Key>::value
+  (::cuda::std::is_arithmetic<Key>::value
+#  ifdef _CCCL_HAS_HALF
```

Comment:
critical: there's no `_CCCL_HAS_HALF`. Half is detected by:
```suggestion
#  ifdef _CCCL_HAS_NVFP16
```

Reply:
Good catch, thx!